segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentContainer.java

```diff
@@ -25,6 +27,18 @@
      */
     Collection<SegmentProperties> getActiveSegments();
 
+    /**
+     * Returns a DirectSegmentAccess object that can be used for operating on a particular StreamSegment directly. The
```

Comment:
I thought the existing Javadoc was explicit enough ... I added more blurb in there.

Reply:
The text is a bit confusing in the following way. The issue (#2855) says the following:

> For a single request, this is the most we can do, but if we have a sequence of requests for the same segment that need to be executed as part of the same external request, then there may be room for optimization.

which talks about multiple requests. I think the `single request` that is mentioned in this javadoc block is the external request because otherwise it seems to be contradicting the purpose of this PR, which is to enable multiple consecutive operations over the same segment. Perhaps you can clarify in the javadoc that you are talking about an external request? Also, if you have a specific example you can share in the javadoc, that would be a nice addition.

Another  question I have out of the javadoc block is based on the following quote:

> If the resulting {@link DirectSegmentAccess} object is cached for more than one request, there is a chance that
 > the underlying Segment may have been evicted from memory.

Say I get an external request X, which translates into two operations A and B via a direct access instance. Is it possible that the underlying segment is evicted from cache between after the execution of A but before the execution of B? If not, why not?